addons/mail/controllers/discuss/channel.py

```diff
@@ -220,3 +221,28 @@ def discuss_delete_sub_channel(self, sub_channel_id):
         channel.parent_channel_id.message_post(body=body, subtype_xmlid="mail.mt_comment")
         # sudo: discuss.channel - skipping ACL for users who created the thread
         channel.sudo().unlink()
+
+    @http.route("/discuss/channel/remove_member", methods=["POST"], type="jsonrpc", auth="public")
+    def discuss_channel_remove_member(self, member_id):
+        channel_member = request.env["discuss.channel.member"].search([("id", "=", member_id)])
+        if not channel_member:
+            raise NotFound()
+        channel = channel_member.channel_id
+        store = Store(bus_channel=channel).delete(channel_member)
+        channel_member.unlink()
+        store.add(channel, fields=["member_count"]).bus_send()
+
+    @http.route("/discuss/channel/set_channel_member_role", methods=["POST"], type="jsonrpc", auth="public")
+    def discuss_channel_set_channel_member_role(self, member_id, channel_role):
+        channel_member = request.env["discuss.channel.member"].search([("id", "=", member_id)])
+        if not channel_member:
+            raise NotFound()
+        channel = channel_member.channel_id
+        # sudo: discuss.channel.member - reading channel role of is considered allowed
+        self_channel_role = channel.self_member_id.sudo().channel_role
+        if not self_channel_role == "owner" and not request.env.user.has_group("base.group_system"):
+            raise AccessError(self.env._("Only the channel owner or DB admins can modify member roles."))
+        if channel_member.sudo().channel_role == "owner" and channel_member != channel.self_member_id and not request.env.user.has_group("base.group_system"):
+            raise AccessError(self.env._("Removing ownership from an owner is not allowed. Please contact your DB administrator."))
```

Comment:
We'll probably need to review all the user facing string with AJU at a later point.